drivers/bluetooth/CMakeLists.txt

```diff
@@ -3,3 +3,7 @@
 zephyr_library()
 zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/bluetooth)
 add_subdirectory(hci)
+
+if(CONFIG_BT_NO_DRIVER)
+    zephyr_library_property(ALLOW_EMPTY TRUE)
```

Comment:
why is this needed ?

It should only be used on old libraries where the amount of work needed to clean up the CMake code was considered to excessive compared to the gain.

Please see this commit message:
https://github.com/zephyrproject-rtos/zephyr/commit/153196b3784032b9dc706c9e2ac3a25e1aedfdbb

> In general libraries should not be created unless there are files to
places inside of it, however in some cases there are so many
`zephyr_library_sources_ifdef(<setting> <file.c>)`
where testing each setting before creating the library may not be
desired.